engine/src/main/scala/cromwell/engine/WorkflowDescriptor.scala

```diff
@@ -239,12 +239,36 @@ object WorkflowDescriptor {
   val CallLogsDirOptionKey = "call_logs_dir"
   val OptionKeys: Set[String] = Set(WorkflowLogDirOptionKey, WorkflowOutputsOptionKey, CallLogsDirOptionKey)
 
+  // This is, frankly, an aberration. But don't worry - it's going away soon :D
+  // TODO: KILL IT WITH FIRE!
+  def getBackendFromOptions(optionsString: String): Backend = {
+    val backendString = WorkflowOptions.fromJsonString(optionsString) flatMap { x => x.get("backend")}
+    backendString match {
+      case Success(x) => CromwellBackend.backend(x)
+      case Failure(_) => CromwellBackend.defaultBackend
+    }
+  }
+
+  /**
+    * Warning: Uses the default backend for this workflow.
+    */
   def apply(id: WorkflowId, sourceFiles: WorkflowSourceFiles): WorkflowDescriptor = {
-    WorkflowDescriptor(id, sourceFiles, ConfigFactory.load)
+    WorkflowDescriptor(id, sourceFiles, WorkflowDescriptor.getBackendFromOptions(sourceFiles.workflowOptionsJson), ConfigFactory.load)
   }
 
+  /**
+    * Warning: Uses the default backend for this workflow.
+    */
   def apply(id: WorkflowId, sourceFiles: WorkflowSourceFiles, conf: Config): WorkflowDescriptor = {
```

Comment:
same question for this `apply`, except it actually is being called from production code in one spot